File: TEST data for large repo/REPO_Copy-6/Blueprism/BluePrism.Automate/BPScheduler/Exceptions/NameAlreadyExistsException.cs
using System;
using System.Runtime.Serialization;

namespace BluePrism.Scheduling
{
    /// <summary>
    /// Exception thrown when a duplicate name within the context
    /// of the scheduler or its ancillary objects.
    /// </summary>
    /// <remarks>
    /// This would be named 'DuplicateNameException' if one with that
    /// name didn't already exist within System.Data specific to data
    /// table names.
    /// </remarks>
    [Serializable]
    public class NameAlreadyExistsException : ScheduleException
    {
        /// <summary>
        /// Creates a new exception with no specific message.
        /// </summary>
        public NameAlreadyExistsException() : base() { }

        /// <summary>
        /// Creates a new exception with the given message.
        /// </summary>
        /// <param name="msg"></param>
        public NameAlreadyExistsException(string msg) : base(msg) { }

        /// <summary>
        /// Creates a new exception with the given formatted message.
        /// </summary>
        /// <param name="msg">The format string for the message. For the
        /// formatting placeholders see <see cref="String.Format"/></param>
        /// <param name="args">The arguments for the specified message.
        /// </param>
        public NameAlreadyExistsException(string msg, params object[] args)
            : base(msg, args) { }

        #region - Serialization handling -

        /// <summary>
        /// Creates and deserializes a name exists exception from the given
        /// serialization objects.
        /// </summary>
        /// <param name="info">The serialized object data</param>
        /// <param name="ctx">The contextual information about the source or
        /// destination</param>
        protected NameAlreadyExistsException(SerializationInfo info, StreamingContext ctx)
            : base(info, ctx) { }

        #endregion

    }
}

File: TEST data for large repo/Repo Copy-3/Blueprism/BluePrism.Server.Domain.Models/Exceptions/NameAlreadyExistsException.vb
Imports System.Runtime.Serialization


''' <summary>
''' Exception thrown when a name is given which is already in use, and the name
''' in question is supposed to be unique.
''' </summary>
<Serializable()> _
Public Class NameAlreadyExistsException : Inherits AlreadyExistsException

    ''' <summary>
    ''' Creates a new exception with no message
    ''' </summary>
    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(message As String)
        MyBase.New(message)
    End Sub

    ''' <summary>
    ''' Creates a new exception with the given formatted message.
    ''' </summary>
    ''' <param name="formattedMsg">The message, with format placeholders, as
    ''' defined in the <see cref="String.Format"/> method.</param>
    ''' <param name="args">The arguments for the formatted message.</param>
    Public Sub New(ByVal formattedMsg As String, ByVal ParamArray args() As Object)
        MyBase.New(formattedMsg, args)
    End Sub

    ''' <summary>
    ''' Creates a new exception from the given de-serializer
    ''' </summary>
    ''' <param name="info">The serialization info from which this exception
    ''' should draw its data.</param>
    ''' <param name="ctx">The context defining the context for the current
    ''' deserialization stream.</param>
    Public Sub New(ByVal info As SerializationInfo, ByVal ctx As StreamingContext)
        MyBase.New(info, ctx)
    End Sub

    Public Sub New(message As String, innerException As Exception)
        MyBase.New(message, innerException)
    End Sub
End Class
